applications/StructuralMechanicsApplication/custom_elements/prestress_membrane_element.cpp

```diff
@@ -1132,142 +1135,552 @@ void PrestressMembraneElement::CalculateMembraneElasticityTensor(
+void PrestressMembraneElement::InitializeSolutionStep(ProcessInfo& rCurrentProcessInfo){
+    // reading integration points and local gradients
+    const GeometryType::IntegrationPointsArrayType& integration_points = GetGeometry().IntegrationPoints();
 
-    // array_1d<double,3> n_pre_pk2=    // Integration of the PreStress over the thickness
+    //const GeometryType::ShapeFunctionsGradientsType& DN_DeContainer = GetGeometry().ShapeFunctionsLocalGradients();
 
-    // the Pre-Stress tensor is defined according to the projection definition proposed by Dr. W�chner
+    //const Matrix& Ncontainer = GetGeometry().ShapeFunctionsValues();
 
-    // creation of T1 ==> Projection of A on the tangential plane
-    CrossProduct(t1, b, g3);
+    // Initialize Variables
+    mdensity = GetProperties()[DENSITY];
+    mThickness = 0.00;
 
-    //if (this->GetId() == 42 && PointNumber == 1)
-    //    KRATOS_WATCH(t1);
+    mTotalDomainInitialSize = 0.00;
 
     
-    // The frame must be orthogonal, so T3 == G3
-    t3 = g3;
+    // resizing jacobian inverses containers
+    mDetJ0.resize(integration_points.size());
+    mGab0.resize(integration_points.size());
+    mG1.resize(integration_points.size());
+    mG2.resize(integration_points.size());
+
+    GeometryType::JacobiansType J0;
+    J0 = GetGeometry().Jacobian(J0);
+
+    //KRATOS_WATCH(J0);
 
-    // Then T2 can only be the cross product between T1 and T3
-    CrossProduct(t2, t3, t1);
+    mStrainsVector.resize(integration_points.size());
+    mStressesVector.resize(integration_points.size());
+    mCauchyStressesVector.resize(integration_points.size()); //VM
 
-    //if (this->GetId() == 42 && PointNumber == 1)
-    //    KRATOS_WATCH(t2);
+    mG_Vector.resize(integration_points.size(), ZeroMatrix(2, 2));
 
+    // initialize prestress matrix and prestress directions
+    Matrix& prestress_variable = this->GetValue(MEMBRANE_PRESTRESS);
+    Matrix& prestress_axis_1 = this->GetValue(PRESTRESS_AXIS_1_GLOBAL);
+    Matrix& prestress_axis_2 = this->GetValue(PRESTRESS_AXIS_2_GLOBAL);
+    unsigned int strain_size = this->GetProperties().GetValue(CONSTITUTIVE_LAW)->GetStrainSize();
 
-    // Normalization of the vectors
-    t1 = t1 / norm_2(t1);
-    t2 = t2 / norm_2(t2);
-    t3 = t3 / norm_2(t3);
 
-    // Transformation matrix from the projected basis T to the local cartesian basis
-    double eG11 = inner_prod(e1, t1);
-    double eG12 = inner_prod(e1, t2);
-    double eG21 = inner_prod(e2, t1);
-    double eG22 = inner_prod(e2, t2);
+    // Calculating geometry tensors in reference configuration on Integration points
+    for (unsigned int PointNumber = 0; PointNumber < integration_points.size(); PointNumber++)
+    {
+        // update prestress (in case of anisotropic prestress)
+        // if step=1: no update necessary
+        if(mAnisotropicPrestress == true && rCurrentProcessInfo[TIME] != 1){
```

Comment:
I wouldn't use the ProcessInfo here. `TIME` can be set to anything by the user
If it is not too expensive then do it all the time, otherwise store a member